base/bucket_gocb.go

```diff
@@ -1386,6 +1586,146 @@ func (bucket CouchbaseBucketGoCB) CouchbaseServerVersion() (major uint64, minor
+	if err != nil {
+		return "", err
+	}
+
+	return goCouchbaseBucket.UUID()
+
+}
+
+func (bucket CouchbaseBucketGoCB) Close() {
+	if err := bucket.Bucket.Close(); err != nil {
+		Warn("Error closing GoCB bucket: %v", err)
+	}
+
+}
+
+// Applies the viewquery options as specified in the params map to the viewQuery object,
+// for example stale=false, etc.
+func applyViewQueryOptions(viewQuery *gocb.ViewQuery, params map[string]interface{}) {
+
+	for optionName, optionValue := range params {
+		switch optionName {
+		case ViewQueryParamStale:
+			optionAsBool := asBool(optionValue)
+			if optionAsBool == false {
+				viewQuery.Stale(gocb.Before)
+			}
+		case ViewQueryParamReduce:
+			viewQuery.Reduce(asBool(optionValue))
+		case ViewQueryParamLimit:
+			uintVal, err := normalizeIntToUint(optionValue)
+			if err != nil {
+				Warn(fmt.Sprintf("%v", err))
+			}
+			viewQuery.Limit(uintVal)
+		case ViewQueryParamDescending:
+			if asBool(optionValue) == true {
+				viewQuery.Order(gocb.Descending)
+			}
+		case ViewQueryParamSkip:
+			uintVal, err := normalizeIntToUint(optionValue)
+			if err != nil {
+				Warn(fmt.Sprintf("%v", err))
+			}
+			viewQuery.Skip(uintVal)
+		case ViewQueryParamGroup:
+			viewQuery.Group(asBool(optionValue))
+		case ViewQueryParamGroupLevel:
+			uintVal, err := normalizeIntToUint(optionValue)
+			if err != nil {
+				Warn(fmt.Sprintf("%v", err))
+			}
+			viewQuery.GroupLevel(uintVal)
+		case ViewQueryParamKey:
+			viewQuery.Key(optionValue)
+		case ViewQueryParamKeys:
+			stringKeys := optionValue.([]string)
+			emptyInterfaceKeys := []interface{}{}
+			for _, key := range stringKeys {
+				emptyInterfaceKeys = append(emptyInterfaceKeys, key)
+			}
+			viewQuery.Keys(emptyInterfaceKeys)
+		}
+
+	}
+
+	// Range: startkey, endkey, inclusiveend
+	var startKey, endKey interface{}
+	if _, ok := params[ViewQueryParamStartKey]; ok {
+		startKey = params[ViewQueryParamStartKey]
+	}
+	if _, ok := params[ViewQueryParamEndKey]; ok {
+		endKey = params[ViewQueryParamEndKey]
+	}
+	inclusiveEnd := false
+	if _, ok := params[ViewQueryParamInclusiveEnd]; ok {
+		inclusiveEnd = asBool(params[ViewQueryParamInclusiveEnd])
+	}
+	viewQuery.Range(startKey, endKey, inclusiveEnd)
+
+	// IdRange: startKeyDocId, endKeyDocId
+	startKeyDocId := ""
+	endKeyDocId := ""
+	if _, ok := params[ViewQueryParamStartKeyDocId]; ok {
+		startKeyDocId = params[ViewQueryParamStartKeyDocId].(string)
+	}
+	if _, ok := params[ViewQueryParamEndKeyDocId]; ok {
+		endKeyDocId = params[ViewQueryParamEndKeyDocId].(string)
+	}
+	viewQuery.IdRange(startKeyDocId, endKeyDocId)
+
+}
+
+func normalizeIntToUint(value interface{}) (uint, error) {
+	switch typeValue := value.(type) {
+	case uint:
+		return typeValue, nil
+	case int:
+		return uint(typeValue), nil
+	case string:
+		i, err := strconv.Atoi(typeValue)
+		return uint(i), err
+	default:
+		return uint(0), fmt.Errorf("Unable to convert %v (%T) -> uint.", value, value)
+	}
+}
```

Comment:
I don't think this handles when value is uint64 (which I believe is what we'll get when someone calls the view endpoint, based on https://github.com/couchbase/sync_gateway/blob/master/rest/handler.go#L407).

If we want this to be fully generic, I think we need all the intn/uintn possibilities, and probably float as well (since JSON unmarshalled numbers end up as float).  It might also make sense to move this into base/util.go.

Do we actually need this to handle any possible int representation for current usage, or is this just to future-proof it against new usages?